drivers/espi/espi_npcx.c

```diff
@@ -387,7 +387,7 @@ static uint32_t espi_taf_parse(const struct device *dev)
 	uint8_t i, roundsize;
 
 	/* Get type, length and tag from RX buffer */
-	memcpy(&taf_head, (void *)&inst->FLASHRXBUF[0], sizeof(taf_head));
+	*(uint32_t *)&taf_head = inst->FLASHRXBUF[0];
```

Comment:
how about casting the right hand side to `struct npcx_taf_head` instead? This is looks a bit weird :-) alternatively I'd suggest adding a local variable for the taf_head pointer so you split this in two

also the commit message should have a "why" for the change, it's not immediately obvious

Reply:
Have updated the code and the commit message.